mava/systems/jax/builder.py

```diff
@@ -114,6 +120,15 @@ def executor(
         self.store.parameter_server_client = parameter_server_client
         self.store.is_evaluator = self.store.executor_id == "evaluator"
 
+        if self.store.is_evaluator:
+            # Set the rng key for the evaluator.
+            self.store.base_key = self.store.eval_key
+        else:
+            # Set the rng key for the executor.
+            self.store.base_key = self.store.executor_keys[
```

Comment:
I see that you did not add the len check for multiple executors here like you do for the trainer since we name the executor with an _index always. I think we should do the same with the trainers in ditributor.py instead of just assigning the name trainer https://github.com/instadeepai/Mava/blob/develop/mava/components/jax/building/distributor.py#L109

Reply:
Good idea, thanks @AsadJeewa :slightly_smiling_face: I updated the code.